crates/wasm/src/shared/implementation/asset.rs

```diff
@@ -0,0 +1,20 @@
+use ambient_std::{
+    asset_cache::SyncAssetKeyExt,
+    asset_url::{AssetUrl, ServerBaseUrlKey, ProxyBaseUrlKey},
+};
+
+use ambient_core::{
+    asset_cache
+};
+
+use ambient_ecs::World;
+
+pub(crate) fn url(
+    world: &mut World,
+    path: String,
+) -> anyhow::Result<Option<String>> {
+    let assets = world.resource(asset_cache());
+    let base_url = ProxyBaseUrlKey.try_get(assets).unwrap_or_else(|| ServerBaseUrlKey.get(assets));
```

Comment:
ProxyBaseUrlKey and ServerBaseUrlKey won't exist on the client side when users join with `ambient join`. They need to be communicated to the client in that case (that's what the `synced_resource` thing I mentioned could be used for).